e2e/e2e-create-windows-nodepool.sh

```diff
@@ -15,16 +15,36 @@ out=$(az aks nodepool list --cluster-name $CLUSTER_NAME -g $RESOURCE_GROUP_NAME
 if [[ "$out" != *"winnp"* ]]; then
     log "Creating windows nodepool"
     if [ "$WINDOWS_E2E_IMAGE" == "2019-containerd" ]; then
-        az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name "winnp" --os-type Windows --node-count 1
+        errorMessage=$(az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name "winnp" --os-type Windows --node-count 1 --only-show-errors 2>&1)
     elif [ "$WINDOWS_E2E_IMAGE" == "2022-containerd" ]; then
-        az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name "winnp" --os-type Windows --os-sku Windows2022 --node-vm-size Standard_D2_v2 --node-count 1
+        errorMessage=$(az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name "winnp" --os-type Windows --os-sku Windows2022 --node-vm-size Standard_D2_v2 --node-count 1 --only-show-errors 2>&1)
     elif [ "$WINDOWS_E2E_IMAGE" == "2022-containerd-gen2" ]; then
-        az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name "winnp" --os-type Windows --os-sku Windows2022 --node-vm-size Standard_DS2_v2 --node-count 1
+        errorMessage=$(az aks nodepool add --resource-group $RESOURCE_GROUP_NAME --cluster-name $CLUSTER_NAME --name "winnp" --os-type Windows --os-sku Windows2022 --node-vm-size Standard_DS2_v2 --node-count 1 --only-show-errors 2>&1)
     else
         err "Invalid windows e2e image version"
         exit 1
     fi
+    if [ "$errorMessage" != "" ]; then
+        log "Other pipeline may be creating the same nodepool, waiting for ready"
+        az aks nodepool wait --nodepool-name "winnp" --cluster-name $CLUSTER_NAME --resource-group $RESOURCE_GROUP_NAME --created --interval 60 --timeout 1800
+        sleep 5
+    fi
     log "Created windows nodepool"
 else
     log "Already create windows nodepool"
 fi
+
+provisioning_state=$(az aks nodepool show --cluster-name $CLUSTER_NAME -g $RESOURCE_GROUP_NAME -n "winnp" -ojson | jq '.provisioningState' | tr -d "\"")
+if [ "$provisioning_state" == "Creating" ]; then
```

Comment:
same here and below to wait it to be Succeeded.

Reply:
check at the end of the file.